### DISCUSSION TITLE
OCCT, GitHub - New way to collaborate [Issues, PRs, Discussions]

### DISCUSSION THREAD

**Discussion by dpasukhi** [Open-Cascade-SAS/OCCT#36] [Announcements]
Hello dear OCCT collaboration forum community,

We are thrilled to announce that the [OCCT GitHub repository](https://github.com/Open-Cascade-SAS/OCCT) is now open for **Pull Request** (PR) integration and **Issue** reporting. For those eager to collaborate with us on GitHub, we have prepared the initial version of our GitHub collaboration workflow. Please note, the current workflow is subject to change, and any updates will be communicated through new forum articles. Documentation will be updated once the working process is fully determined and described.

**OCCT GitHub Workflow (V1):**

1. **Licensing:** OCCT is available under the LGPL 2.1 license. To integrate any of your changes into our repositories, collaborators need to sign the Contributor License Agreement (CLA). Details on how to sign and submit the CLA are available on the following page [Get Involded](https://dev.opencascade.org/get_involved)].

2. **Bug Tracking:**
**Information for Mantis repository**. All OCCT commits must be associated with a specific bug ticket in our bug tracker. We recommend creating a profile in the bug tracker [BugTracker](https://tracker.dev.opencascade.org/)] when you sign the CLA. This is required for integrating your contributions into the OCCT repository.
**Information for GitHub repository**. GitHub users can describe their issue in a special PR together with code. Additionally, all GitHub users can report issue with any template. OCCT team will update PR or Issue manually.

4. **Forking:** PRs to the OCCT repository are accepted only from external forks. Please create your own fork and work within your repository during the development process.

5. **PR Submission:** Once you have completed your code updates and prepared the final fix, please submit a PR to the OpenCascade OCCT repository using the following templates:

    - **If your PR has a connected bug ticket in our bug tracker:**
      - **PR Name Template:** `<ticket number>: <Category(es)> - <Summary description>`
      - **PR Description:** Should contain a summary of your changes in a clear and concise manner.
      - **Example:**
        ```
        0032750: Visualization, AIS_Manipulator - selection of moved object is broken
        Completed the stop transform action when dragging manipulator with mouse.
        Added context redisplay for update of interactive object sensitive areas.
        Added test.
        ```

    - **If your PR does not have a bug ticket:**
      - **PR Name Template:** `<Category(es)> - <Summary description>`
      - (If related GitHub issue is exist, needs to connect using GitHub UI)
      - **PR Description:** Should include the following details:
        1. Steps to reproduce the issue.
        2. Any necessary files related to the bug.
        3. Version of OCCT when the bug was found.
        4. Type of the issue (minor, major, crash, documentation).
        6. Platform where the issue was found or fixed.

    The information about the fix itself should be presented in one of your commits. The git commit message should match the PR template for cases when you already have a created bug ticket.

5. **CLA Number:** Please include your CLA number in the GitHub note. This number is provided in an email from us that confirms your CLA signing request. This step is required only once per account.

7. **Branching:** PRs to the master branch are not accepted. After verifying your CLA, the OCCT team will update the PR base branch to an IR branch. The IR branch is used to collect a sequence of git commits for integration into the master branch, helping us validate and test issues collectively to avoid conflicts.

8. **Testing:** The OCCT team will initiate the testing process and share the results with you. If there are any regressions, we will ask you to address them. If testing on your end is not possible, the OCCT team will take over to fix the regressions. In such cases, the issue will be queued for the next available developer.

9. **Code Review:** The OCCT team will conduct a code review, sharing any remarks and suggestions to ensure a collaborative integration process. Any changes during the code review should be made by the PR submitter.

10. **Approval and Merging:** Upon successful testing, an OCCT team member will approve your PR, and the bugmaster will merge your PR into the IR-n branch.

11. **IR-n Branch Lifecycle:** The IR-n branch will exist for a period during the internal development cycle. You will be notified when your changes have been merged with the master branch.

12. **GitHub Issue and Discussions**. All GitHub users can submit the issue or new discussion. OCCT Team will update the information or request more details from reporter.

We are committed to simplifying the collaboration experience. Your suggestions and feedback are always welcome.

Best regards,
OCCT3d development team.

**Comment by kadet1090** [comment]
Would it be possible to simplify CLA signing process? Mostly asking to provide some way of signing it via online means which will make it easier for new people to contribute. Microsoft has such process for VS Code, there exist stuff like: https://github.com/cla-assistant/cla-assistant to help with that.

**Reply by dpasukhi** [reply]
I hope it is possible, but it requires a lot of internal updates and approvals.
Everything related to law is very sensitive in the corporate world.
I initiated a task to reorganize that approach, but it will take some time (probably a long time).
When there is any news, I will share it.

Thank you for your suggestion! That example is very interesting.

Best regards,
Dmitrii

**Comment by ahoarau** [comment]
@dpasukhi Very good initiative ! This will definitely make collaboration much smoother.

**Comment by dpasukhi** [comment]
A new extension for collaboration - "Issues" based on GitHub. You are free to share your problems in GitHub without external bug systems.

The link: [Issues · Open-Cascade-SAS/OCCT (github.com)](https://github.com/Open-Cascade-SAS/OCCT/issues)

Best regards, OCCT3D Team.
